src/util/std_code.cpp

```diff
@@ -167,3 +170,57 @@ void code_function_bodyt::set_parameter_identifiers(
     sub.back().set(ID_identifier, id);
   }
 }
+
+code_inputt::code_inputt(
+  std::vector<exprt> arguments,
+  optionalt<source_locationt> location)
+  : codet{ID_input, std::move(arguments)}
+{
+  if(location)
+    add_source_location() = std::move(*location);
+}
+
+code_inputt::code_inputt(
+  const irep_idt &identifier,
+  exprt expression,
+  optionalt<source_locationt> location)
+  : code_inputt{{address_of_exprt(index_exprt(
+                   string_constantt(identifier),
+                   from_integer(0, index_type()))),
+                 std::move(expression)},
+                std::move(location)}
+{
+}
+
```

Comment:
It's a bit odd that you have to choose between multiple inputs in a vector with no description and one input with a description.

Reply:
The constructor which takes one input with a description is a convenient interface for use in the construction of `cprover_start` inside cbmc/jbmc. Its existence means that we can reduce the amount of duplication in the code which produces the various different versions of `cprover_start`.

The constructor which takes multiple inputs in a vector is required in order to support arbitrary calls to `__CPROVER_input` in user code. The definition in `cprover.h `is `void __CPROVER_input(const char *description, ...);`. This would allow the first parameter to be any pointer to `char`, rather than a string literal only. The `...` would allow any number of expressions to be associated with the description, rather than a single expression only. The existing invariants only constrain the `...` to correspond to at least one expression. Therefore this additional constructor cannot be removed without further constraining what user code is considered valid. My intention was to create a higher level internal interface, rather than make externally facing changes to existing functionality.

Is my explanation of this aspect sufficient, or do you want me to make further changes to the code in the PR - such documenting the purposes of the two constructors?